python/dolfinx/fem/function.py

```diff
@@ -431,26 +432,28 @@ def __init__(self, mesh: Mesh, element: typing.Union[ufl.FiniteElementBase, Elem
             self._cpp_object = cppV
             return
 
-        # Initialise the ufl.FunctionSpace
-        if isinstance(element, ufl.FiniteElementBase):
-            super().__init__(mesh.ufl_domain(), element)
-        else:
-            e = ElementMetaData(*element)
-            ufl_element = ufl.FiniteElement(e.family, mesh.ufl_cell(), e.degree, form_degree=e.form_degree)
-            super().__init__(mesh.ufl_domain(), ufl_element)
+        # This if is required for type checks to pass
```

Comment:
What precisely is required?

Reply:
This part of the code must be guarded by this if statement in order that the static type checker does not check it for the `None` type (which fails to pass).